src/site/content/en/blog/google-search-sw/index.md

```diff
@@ -0,0 +1,463 @@
+there should not be any latency overhead introduced by the service worker.
+
+![An illustration of the SW startup done in parallel with the navigation request.](with-preload.png)
+
+The Search team also needed to avoid using a service worker on low-end mobile
+devices where that service worker boot time could exceed the time it would take
+to get back a network response. Since there's no hard-and-fast rule for what
+constitutes a "low-end" device, they came up with the heuristic of
+[checking the total RAM](https://developers.google.com/web/updates/2017/12/device-memory)
+installed on the device. Anything less than 2 gigabytes of memory fell into
+their low-end device category, where based on the real-world experiments they've
+run, service worker startup time would be unacceptable.
+
+Available storage space is another consideration, since the full set of
+resources to be cached for future use can run to several megabytes. The
+[`navigator.storage` interface](https://developers.google.com/web/updates/2017/08/estimating-available-storage-space)
+comes in handy there, allowing the Google Search page to figure out in advance
+whether their attempts to cache data run the risk of failing due to storage
+quota failures.
+
+This left the Search team with multiple pieces of criteria that they could use
+to determine whether or not to use a service worker: if a user comes to the
+Google Search page using a browser that supports navigation preload, and has at
+least 2 gigabytes of RAM, and enough free storage space, then a
+[service worker will be registered](https://developers.google.com/web/fundamentals/primers/service-workers/lifecycle#the_first_service_worker).
+Browsers or devices that don't meet that criteria won't end up with a service
+worker, but they'll still see the same Google Search experience as they always
+have.
+
+One side benefit of this selective registration is that it has allowed the
+Search team to ship a smaller, more efficient service worker. Knowing that only
+fairly modern browsers will end up running the service worker code means
+eliminating the overhead of transpilation and polyfills needed to target
+browsers on which registration will never take place. This ended up cutting out
+around 8 kilobytes of uncompressed JavaScript code from the total size of the
+service worker's implementation.
+
+### Service worker scopes
+
+Once the Search team ran enough latency experiments and were confident that
+using navigation preload offered them a viable, latency-neutral path for using a
+service worker, some practical issues started moving to the forefront. One of
+those issues has to do with service worker's
+[scoping rules](https://developers.google.com/web/ilt/pwa/introduction-to-service-worker#registration_and_scope).
+A service worker's scope determines which pages it can potentially take control
+of, and scoping works based on URL path prefix. For domains that host a single
```

Comment:
Suggestion to break up sentence and start a new paragraph. 

A service worker's scope determines which pages it can potentially take control of.

New paragraph.

Scoping works based on the URL path prefix. For domains...